lib/elixir/lib/config.ex

```diff
@@ -62,7 +62,25 @@ defmodule Config do
         import_config config
       end
 
-  The last step is to replace all `Mix.env()` calls by `config_env()`.
+  The last step is to replace all `Mix.env()` calls in the config files with `config_env()`.
+
+  To check an application's environment at runtime, you may add a configuration key:
+
+      # config.exs
+      ...
+      config :my_app, env: config_env()
+
+  Then, in `mix.exs` and other files, you may get the environment via:
```

Comment:
You can still use `Mix.env()` in `mix.exs` so I would suggest another example.

Reply:
I attempted to clarify things a bit in the newest commit.

Reply:
to add Eric's comment, the section you are updating is about migrating `use Mix.Config` to `import Config` and thus the extra information is a bit offtopic _there_.

I feel like mentioning the env outside of `config.exs` is a bit of a different problem, basically how to avoid using `Mix.env()` at _runtime_, isn't it? If so, as Eric mentioned it is fine to use it in `mix.exs`. Sorry if the following is obvious, it is also fine to use it in module _bodies_. This is fine:

```elixir
defmodule Foo do
  @env Mix.env()

  def foo do
    @env
  end
end
```

and so is:

```elixir
defmodule Foo do
  def foo do
    unquote(Mix.env())
  end
end
```

Reply:
ahh ok. I was misunderstanding that `Mix.env()` was still ok in modules. The line in the docs says to replace all instances of `Mix.env()`.

Would it make sense to say something like this, based on the current docs?

> The last step is to replace all Mix.env() calls in the configuration files with config_env(). Elsewhere in the application, Mix.env() is still useful for identifying the environment at runtime.

Reply:
> Mix.env() is still useful for identifying the environment at runtime

this is not quite accurate. we shouldn't call `Mix.env()` at runtime, that is directly in the function body, we should only call it at compile time, that is in module body or in unquote.

Btw, I just looked up [`Mix.env/0`](https://hexdocs.pm/mix/Mix.html#env/0) docs and we have:

> This function should not be used at runtime in application code (as opposed to infrastructure and build code like Mix tasks). Mix is a build tool and may not be available after the code is compiled (for example in a release).

> To differentiate the program behavior depending on the environment, it is recommended to use application environment through Application.get_env/3. Proper configuration can be set in config files, often per-environment (see the Config module for more information).

so more in line with what you had in mind after all.

Maybe in the section you're changing we just leave a pointer to `Mix.env()`?

Reply:
I'd argue that a note about Mix.env() would be in scope for the section about migrating from Mix.Config. I prefer a couple small concrete examples of Mix.env() use, but I understand you don't want to explain runtime vs compile time if it's been done elsewhere. Still, I know this will trip me up in the future, possibly others.